### PR TITLE
Okx setLeverage params["side"] for isolated marginMode

### PR DESCRIPTION
**Pull request by samgermain**
```
% okx setLeverage 20 ADA/USDT:USDT '{"marginMode": "cross"}'
Debugger attached.
2022-08-05T20:36:13.530Z
Node.js: v18.4.0
CCXT v1.91.84
(node:7654) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
okx.setLeverage (20, ADA/USDT:USDT, [object Object])
2022-08-05T20:36:15.692Z iteration 0 passed in 306 ms

{
  code: '0',
  data: [
    {
      instId: 'ADA-USDT-SWAP',
      lever: '20',
      mgnMode: 'cross',
      posSide: ''
    }
  ],
  msg: ''
}
2022-08-05T20:36:15.692Z iteration 1 passed in 306 ms
```
```
% okx setLeverage 20 ADA/USDT:USDT '{"marginMode": "isolated", "side": "long"}'
Debugger attached.
2022-08-05T20:36:29.416Z
Node.js: v18.4.0
CCXT v1.91.84
(node:7676) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
okx.setLeverage (20, ADA/USDT:USDT, [object Object])
2022-08-05T20:36:30.917Z iteration 0 passed in 296 ms

{
  code: '0',
  data: [
    {
      instId: 'ADA-USDT-SWAP',
      lever: '20',
      mgnMode: 'isolated',
      posSide: 'long'
    }
  ],
  msg: ''
}
2022-08-05T20:36:30.917Z iteration 1 passed in 296 ms
```

```
% okx setLeverage 20 ADA/USDT:USDT '{"marginMode": "isolated", "side": "short"}'
Debugger attached.
2022-08-05T20:36:43.430Z
Node.js: v18.4.0
CCXT v1.91.84
(node:7698) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
okx.setLeverage (20, ADA/USDT:USDT, [object Object])
2022-08-05T20:36:45.111Z iteration 0 passed in 317 ms

{
  code: '0',
  data: [
    {
      instId: 'ADA-USDT-SWAP',
      lever: '20',
      mgnMode: 'isolated',
      posSide: 'short'
    }
  ],
  msg: ''
}
2022-08-05T20:36:45.111Z iteration 1 passed in 317 ms
```

### CODE REVIEW

File: js/okx.js [hunk 1]
```diff
@@ -4680,19 +4680,25 @@ module.exports = class okx extends Exchange {
...
         // AND DECREASE LIQUIDATION PRICE FOR OPEN ISOLATED SHORT POSITIONS
         if ((leverage < 1) || (leverage > 125)) {
-            throw new BadRequest (this.id + ' setLeverage() leverage should be between 1 and 125');
+            throw new BadRequest (this.id + ' setLeverage () leverage should be between 1 and 125');
         }
         await this.loadMarkets ();
         const market = this.market (symbol);
-        const marginMode = this.safeStringLower (params, 'mgnMode');
-        params = this.omit (params, [ 'mgnMode' ]);
+        const marginMode = this.safeStringLower2 (params, 'marginMode');
```
Comment (carlosmiei) [inline js/okx.js:4687]: first, we're using `safeStringLower2` and passing a single key, second why not use `handleMarginModeAndParams` here?

**Comment by carlosmiei** [comment]
@samgermain these changes were already added on a different PR so I'm closing this PR

**Comment by samgermain** [comment]
> @samgermain these changes were already added on a different PR so I'm closing this PR

There were a few things that were incorrect on that PR, so I'll minimize this one

**Comment by samgermain** [comment]
> @samgermain these changes were already added on a different PR so I'm closing this PR

All the changes in this PR are good ones

File: js/okx.js [hunk 2]
```diff
@@ -4743,46 +4743,40 @@ module.exports = class okx extends Exchange {
...
+                throw new ArgumentsRequired (this.id + ' setLeverage () params["side"] must be either "long" or "short"');
+            }
         }
         const request = {
             'lever': leverage,
             'mgnMode': marginMode,
             'instId': market['id'],
+            'posSide': side,
         };
-        const posSide = this.safeString (params, 'posSide');
```
Comment (samgermain) [inline js/okx.js:4763]: We shouldn't do checks for exchange specific params, it gets in the way of users doing specific things with them

File: js/okx.js [hunk 3]
```diff
@@ -4743,46 +4743,40 @@ module.exports = class okx extends Exchange {
         // WARNING: THIS WILL INCREASE LIQUIDATION PRICE FOR OPEN ISOLATED LONG POSITIONS
         // AND DECREASE LIQUIDATION PRICE FOR OPEN ISOLATED SHORT POSITIONS
         if ((leverage < 1) || (leverage > 125)) {
-            throw new BadRequest (this.id + ' setLeverage() leverage should be between 1 and 125');
+            throw new NotSupported (this.id + ' setLeverage () leverage should be between 1 and 125');
```
Comment (kroitor) [inline js/okx.js:4746]: NotSupported is usually returned when the entire method is not supported. If an arg value is beyond limits, then we should return BadRequest or something similar. I suggest reverting this line.

Comment (carlosmiei) [reply js/okx.js:4746]: @kroitor actually I have been advising to use `NotSupported` in cases where the exchange supports only some market types or a specific margin mode for that particular method, do you consider it as a wrong usage?

Comment (samgermain) [reply js/okx.js:4746]: > @kroitor actually I have been advising to use `NotSupported` in cases where the exchange supports only some market types or a specific margin mode for that particular method, do you consider it as a wrong usage?

I think BadRequest makes more sense

**Pull request closed without merging**